buildrunner/docker/runner.py

```diff
@@ -323,32 +328,33 @@ def restore_caches(self, logger: ContainerLogger, caches: OrderedDict) -> None:
                 continue
 
             orig_shell = self.shell
+            file_obj = None
             try:
                 self.shell = "/bin/sh"
                 exit_code = self.run(f"mkdir -p {docker_path}")
                 if exit_code:
                     logger.write(f"WARNING: There was an issue creating {docker_path} on the docker container\n")
 
-                with open(actual_cache_archive_file, 'rb') as data:
-                    try:
-                        # Allow multiple people to read from the file at the same time
-                        # If another instance adds an exclusive lock in the save method below,
-                        # this will block until the lock is released
-                        fcntl.lockf(data, fcntl.LOCK_SH)
-                        restored_cache_src.add(docker_path)
-                        if not self.docker_client.put_archive(self.container['Id'], docker_path, data):
-                            logger.write(
-                                f"WARNING: An error occurred when trying to use cache "
-                                f"{actual_cache_archive_file} at the path {docker_path}\n"
-                            )
-                    finally:
-                        # Always unlock the file after extracting the archive
-                        fcntl.lockf(data, fcntl.LOCK_UN)
+                # Allow multiple people to read from the file at the same time
+                # If another instance adds an exclusive lock in the save method below,
+                # this will block until the lock is released
+                file_obj = acquire_read_binary_flock(
+                    lock_file=actual_cache_archive_file,
+                    logger=logger,
+                    timeout_seconds=600
```

Comment:
After thinking on this more, I think 10 minutes is pretty long. Maybe we should do 5?

Reply:
Agreed. I'll make the change.